### PR TITLE
Pyth network - user updating content to Amoy network

### PR DESCRIPTION
**Pull request by aditya520**

### CODE REVIEW

**Comment by kmurphypolygon** [comment]
Hi @aditya520 thanks so much for your submission. Could you add a menu item please otherwise your doc will not be visible on the site. Follow the README instructions to get spun up. Thank you :)

File: docs/tools/oracles/pyth.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
+# Pyth Network
```
Comment (kmurphypolygon) [inline docs/tools/oracles/pyth.md]:
```suggestion
# Pyth network
```

File: docs/tools/oracles/pyth.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+[Pyth Network]: https://pyth.network/
+
+## Overview
+
+The [Pyth Network] is the **largest** first-party financial oracle network, delivering real-time market data to over 40 blockchains securely and transparently.
+
+The network comprises some of the world’s largest exchanges, market makers, and financial services providers publishing their proprietary price data on-chain for aggregation and distribution to smart contract applications.
+
+## Using Pyth Network
```
Comment (kmurphypolygon) [inline docs/tools/oracles/pyth.md]:
```suggestion
## Using Pyth network
```

File: docs/tools/oracles/pyth.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
+# Pyth network
+
+[Pyth Network]: https://pyth.network/
+
+## Overview
+
+The [Pyth Network] is the **largest** first-party financial oracle network, delivering real-time market data to over 40 blockchains securely and transparently.
+
+The network comprises some of the world’s largest exchanges, market makers, and financial services providers publishing their proprietary price data on-chain for aggregation and distribution to smart contract applications.
```
Comment (kmurphypolygon) [inline docs/tools/oracles/pyth.md]:
```suggestion
The [Pyth network](https://pyth.network/) is the largest first-party financial oracle network, delivering real-time market data to over 40 blockchains securely and transparently.

The network comprises some of the world’s largest exchanges, market makers, and financial services providers publishing their proprietary price-data on-chain for aggregation and distribution to smart contract applications.
```

File: docs/tools/oracles/pyth.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
+# Pyth network
+
+[Pyth Network]: https://pyth.network/
```
Comment (kmurphypolygon) [inline docs/tools/oracles/pyth.md]:
```suggestion
```

File: docs/tools/oracles/pyth.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
+# Pyth network
```
Comment (kmurphypolygon) [inline docs/tools/oracles/pyth.md]:
```suggestion
```

File: docs/tools/oracles/pyth.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+[Pyth Network]: https://pyth.network/
+
```
Comment (kmurphypolygon) [inline docs/tools/oracles/pyth.md]:
```suggestion
```

File: docs/tools/oracles/pyth.md [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+The network comprises some of the world’s largest exchanges, market makers, and financial services providers publishing their proprietary price-data on-chain for aggregation and distribution to smart contract applications.
+
+## Using Pyth network
+                                                                                       
+The Pyth Network introduced an innovative low-latency [pull oracle design](https://docs.pyth.network/documentation/pythnet-price-feeds/on-demand), where users are empowered to “pull” price updates on-chain when needed, enabling everyone in that blockchain environment to access that data point. 
+
+Developers on Polygon have permissionless access to any of Pyth’s 350+ price feeds for equities, ETFs, commodities, foreign exchange pairs, and cryptocurrencies.
+
+Here is a working example of a contract that fetches the latest price of MATIC/USD on the Polygon mainnet. 
+You have to pass [Pyth's contract address](https://docs.pyth.network/price-feeds/contract-addresses/evm) of Polygon mainnet/mumbai and the desired [price feed](https://pyth.network/developers/price-feed-ids) id](https://pyth.network/developers/price-feed-ids) to fetch the latest price
```
Comment (kmurphypolygon) [inline docs/tools/oracles/pyth.md]:
```suggestion
The Pyth network introduces an innovative low-latency [pull oracle design](https://docs.pyth.network/documentation/pythnet-price-feeds/on-demand), where users can pull price updates on-chain when needed, enabling everyone in that blockchain environment to access that data point. 

Developers on Polygon have permissionless access to any of Pyth’s 350+ price feeds for equities, ETFs, commodities, foreign exchange pairs, and cryptocurrencies.

Here is a working example of a contract that fetches the latest price of MATIC/USD on the Polygon mainnet. 
You have to pass [Pyth's contract address](https://docs.pyth.network/price-feeds/contract-addresses/evm) for Polygon mainnet/mumbai and the desired [price feed](https://pyth.network/developers/price-feed-ids) id](https://pyth.network/developers/price-feed-ids) to fetch the latest price.
```

**Comment by kmurphypolygon** [comment]
Hi @aditya520 I notice you are documenting against the Polygon Mumbai network which we have just deprecated. Please could you update the doc for the Amoy network. Thank you so much :)

**Comment by aditya520** [comment]
@kmurphypolygon Can you please verify and merge.

**Comment by kmurphypolygon** [comment]
Hi @aditya520, thank you so much, I'll merge and push this live asap.

File: docs/tools/oracles/pyth.md [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+The network comprises some of the world’s largest exchanges, market makers, and financial services providers publishing their proprietary price-data on-chain for aggregation and distribution to smart contract applications.
+
+## Using Pyth network
+                                                                                       
+The Pyth network introduces an innovative low-latency [pull oracle design](https://docs.pyth.network/documentation/pythnet-price-feeds/on-demand), where users can pull price updates on-chain when needed, enabling everyone in that blockchain environment to access that data point. 
+
+Developers on Polygon have permissionless access to any of Pyth’s 350+ price feeds for equities, ETFs, commodities, foreign exchange pairs, and cryptocurrencies.
+
+Here is a working example of a contract that fetches the latest price of MATIC/USD on the Polygon network. 
+You have to pass [Pyth's contract address](https://docs.pyth.network/price-feeds/contract-addresses/evm) for Polygon mainnet/amoy and the desired [price feed](https://pyth.network/developers/price-feed-ids) id](https://pyth.network/developers/price-feed-ids) to fetch the latest price.
```
Comment (kmurphypolygon) [inline docs/tools/oracles/pyth.md]:
```suggestion
Here is a working example of a contract that fetches the latest price of MATIC/USD on the Polygon network. 
You have to pass [Pyth's contract address](https://docs.pyth.network/price-feeds/contract-addresses/evm) for Polygon mainnet/amoy and the desired [price feed id](https://pyth.network/developers/price-feed-ids) to fetch the latest price.
```

File: docs/tools/oracles/pyth.md [hunk 9]
```diff
@@ -0,0 +1,68 @@
+## Overview
+
+The [Pyth network](https://pyth.network/) is the largest first-party financial oracle network, delivering real-time market data to over 40 blockchains securely and transparently.
+
+The network comprises some of the world’s largest exchanges, market makers, and financial services providers publishing their proprietary price-data on-chain for aggregation and distribution to smart contract applications.
```
Comment (kmurphypolygon) [inline docs/tools/oracles/pyth.md]:
```suggestion
The network comprises some of the world’s largest exchanges, market makers, and financial services providers; publishing proprietary price-data on-chain for aggregation and distribution to smart contract applications.
```

File: docs/tools/oracles/pyth.md [hunk 10]
```diff
@@ -0,0 +1,68 @@
+## Overview
+
+The [Pyth network](https://pyth.network/) is the largest first-party financial oracle network, delivering real-time market data to over 40 blockchains securely and transparently.
+
+The network comprises some of the world’s largest exchanges, market makers, and financial services providers; publishing proprietary price-data on-chain for aggregation and distribution to smart contract applications.
+
+## Using Pyth network
+                                                                                       
+The Pyth network introduces an innovative low-latency [pull oracle design](https://docs.pyth.network/documentation/pythnet-price-feeds/on-demand), where users can pull price updates on-chain when needed, enabling everyone in that blockchain environment to access that data point.
```
Comment (kmurphypolygon) [inline docs/tools/oracles/pyth.md]:
```suggestion
The Pyth network introduces an innovative low-latency [pull oracle design](https://docs.pyth.network/documentation/pythnet-price-feeds/on-demand), where users can pull price updates on-chain when needed, enabling everyone in the blockchain environment to access that data point. 
```

File: docs/tools/oracles/pyth.md [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+The network comprises some of the world’s largest exchanges, market makers, and financial services providers; publishing proprietary price-data on-chain for aggregation and distribution to smart contract applications.
+
+## Using Pyth network
+                                                                                       
+The Pyth network introduces an innovative low-latency [pull oracle design](https://docs.pyth.network/documentation/pythnet-price-feeds/on-demand), where users can pull price updates on-chain when needed, enabling everyone in the blockchain environment to access that data point. 
+
+Developers on Polygon have permissionless access to any of Pyth’s 350+ price feeds for equities, ETFs, commodities, foreign exchange pairs, and cryptocurrencies.
+
+Here is a working example of a contract that fetches the latest price of MATIC/USD on the Polygon network. 
+You have to pass [Pyth's contract address](https://docs.pyth.network/price-feeds/contract-addresses/evm) for Polygon mainnet/amoy and the desired [price feed id](https://pyth.network/developers/price-feed-ids) to fetch the latest price.
```
Comment (kmurphypolygon) [inline docs/tools/oracles/pyth.md]:
```suggestion
You have to pass [Pyth's contract address](https://docs.pyth.network/price-feeds/contract-addresses/evm) for Polygon mainnet/testnet and the desired [price feed id](https://pyth.network/developers/price-feed-ids) to fetch the latest price.
```

File: docs/tools/oracles/pyth.md [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+
+        // Fetch the latest price
+        PythStructs.Price memory price = pyth.getPrice(priceFeed);
+        return price.price;
+    }
+}
+
+```
+
+Here you can fetch the `updateData` from our [`Hermes`](https://docs.pyth.network/price-feeds/pythnet-price-feeds/hermes), which listens to Pythnet and Wormhole for price updates or you can use [`pyth-evm-js`](https://github.com/pyth-network/pyth-crosschain/blob/main/target_chains/ethereum/sdk/js/src/EvmPriceServiceConnection.ts#L15) sdk.
```
Comment (kmurphypolygon) [inline docs/tools/oracles/pyth.md]:
```suggestion
Here you can fetch the `updateData` from our [`Hermes` feed](https://docs.pyth.network/price-feeds/pythnet-price-feeds/hermes), which listens to Pythnet and Wormhole for price updates; or you can use the [`pyth-evm-js`](https://github.com/pyth-network/pyth-crosschain/blob/main/target_chains/ethereum/sdk/js/src/EvmPriceServiceConnection.ts#L15) SDK. 
```

File: docs/tools/oracles/pyth.md [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+        return price.price;
+    }
+}
+
+```
+
+Here you can fetch the `updateData` from our [`Hermes` feed](https://docs.pyth.network/price-feeds/pythnet-price-feeds/hermes), which listens to Pythnet and Wormhole for price updates; or you can use the [`pyth-evm-js`](https://github.com/pyth-network/pyth-crosschain/blob/main/target_chains/ethereum/sdk/js/src/EvmPriceServiceConnection.ts#L15) SDK. 
+
+
+This [package](https://github.com/pyth-network/pyth-crosschain/tree/main/target_chains/ethereum/sdk/solidity) provides utilities for consuming prices from the Pyth Network Oracle using Solidity. Also, it contains the [Pyth Interface ABI](https://github.com/pyth-network/pyth-crosschain/blob/main/target_chains/ethereum/sdk/solidity/abis/IPyth.json) that you can use in your libraries to communicate with the Pyth contract.
```
Comment (kmurphypolygon) [inline docs/tools/oracles/pyth.md]:
```suggestion
This [package](https://github.com/pyth-network/pyth-crosschain/tree/main/target_chains/ethereum/sdk/solidity) provides utilities for consuming prices from the Pyth network oracle using Solidity. Also, it contains the [Pyth Interface ABI](https://github.com/pyth-network/pyth-crosschain/blob/main/target_chains/ethereum/sdk/solidity/abis/IPyth.json) that you can use in your libraries to communicate with the Pyth contract.
```

File: docs/tools/oracles/pyth.md [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+}
+
+```
+
+Here you can fetch the `updateData` from our [`Hermes` feed](https://docs.pyth.network/price-feeds/pythnet-price-feeds/hermes), which listens to Pythnet and Wormhole for price updates; or you can use the [`pyth-evm-js`](https://github.com/pyth-network/pyth-crosschain/blob/main/target_chains/ethereum/sdk/js/src/EvmPriceServiceConnection.ts#L15) SDK. 
+
+
+This [package](https://github.com/pyth-network/pyth-crosschain/tree/main/target_chains/ethereum/sdk/solidity) provides utilities for consuming prices from the Pyth network oracle using Solidity. Also, it contains the [Pyth Interface ABI](https://github.com/pyth-network/pyth-crosschain/blob/main/target_chains/ethereum/sdk/solidity/abis/IPyth.json) that you can use in your libraries to communicate with the Pyth contract.
+
+It is strongly recommended to follow the consumer's [best practices](https://docs.pyth.network/documentation/pythnet-price-feeds/best-practices) when consuming Pyth data.
```
Comment (kmurphypolygon) [inline docs/tools/oracles/pyth.md]:
```suggestion
We recommend following the [consumer best practices](https://docs.pyth.network/documentation/pythnet-price-feeds/best-practices) when consuming Pyth data. 
```

File: docs/tools/oracles/pyth.md [hunk 15]
```diff
@@ -0,0 +1,68 @@
...
+```
+
+Here you can fetch the `updateData` from our [`Hermes` feed](https://docs.pyth.network/price-feeds/pythnet-price-feeds/hermes), which listens to Pythnet and Wormhole for price updates; or you can use the [`pyth-evm-js`](https://github.com/pyth-network/pyth-crosschain/blob/main/target_chains/ethereum/sdk/js/src/EvmPriceServiceConnection.ts#L15) SDK. 
+
+
+This [package](https://github.com/pyth-network/pyth-crosschain/tree/main/target_chains/ethereum/sdk/solidity) provides utilities for consuming prices from the Pyth network oracle using Solidity. Also, it contains the [Pyth Interface ABI](https://github.com/pyth-network/pyth-crosschain/blob/main/target_chains/ethereum/sdk/solidity/abis/IPyth.json) that you can use in your libraries to communicate with the Pyth contract.
+
+We recommend following the [consumer best practices](https://docs.pyth.network/documentation/pythnet-price-feeds/best-practices) when consuming Pyth data. 
+
+For more information and details, please refer to the official documentation [here](https://docs.pyth.network/price-feeds).
```
Comment (kmurphypolygon) [inline docs/tools/oracles/pyth.md]:
```suggestion
For more information, check out the official [Pyth documentation](https://docs.pyth.network/price-feeds).
```

File: docs/tools/oracles/pyth.md [hunk 16]
```diff
@@ -0,0 +1,68 @@
...
+
+
+This [package](https://github.com/pyth-network/pyth-crosschain/tree/main/target_chains/ethereum/sdk/solidity) provides utilities for consuming prices from the Pyth network oracle using Solidity. Also, it contains the [Pyth Interface ABI](https://github.com/pyth-network/pyth-crosschain/blob/main/target_chains/ethereum/sdk/solidity/abis/IPyth.json) that you can use in your libraries to communicate with the Pyth contract.
+
+We recommend following the [consumer best practices](https://docs.pyth.network/documentation/pythnet-price-feeds/best-practices) when consuming Pyth data. 
+
+For more information, check out the official [Pyth documentation](https://docs.pyth.network/price-feeds).
+
+You can find more details on the various functions available to you when interacting with the Pyth smart contract in the [API Reference section](https://docs.pyth.network/price-feeds/api-reference/evm).
```
Comment (kmurphypolygon) [inline docs/tools/oracles/pyth.md]:
```suggestion
For more information, check out the official [Pyth documentation](https://docs.pyth.network/price-feeds). There are details on the various functions available for interacting with the Pyth smart contract in the [API Reference section](https://docs.pyth.network/price-feeds/api-reference/evm).
```

File: docs/tools/oracles/pyth.md [hunk 17]
```diff
@@ -0,0 +1,66 @@
...
+For more information, check out the official [Pyth documentation](https://docs.pyth.network/price-feeds). There are details on the various functions available for interacting with the Pyth smart contract in the [API Reference section](https://docs.pyth.network/price-feeds/api-reference/evm).
+
+## Pyth on Polygon
+
+The Pyth Network smart contract is available at the following address: 
+
+- Mainnet: [0xff1a0f4744e8582DF1aE09D5611b887B6a12925C](https://polygonscan.com/address/0xff1a0f4744e8582df1ae09d5611b887b6a12925c).
+- Amoy: [0x2880aB155794e7179c9eE2e38200202908C17B43](https://www.oklink.com/amoy/address/0x2880ab155794e7179c9ee2e38200202908c17b43)
+
+Additionally, you'll be able to find all the Pyth Price Feed IDs [here](https://pyth.network/developers/price-feed-ids).
```
Comment (kmurphypolygon) [inline docs/tools/oracles/pyth.md]:
```suggestion
Additionally, click to access the [Pyth price-feed IDs](https://pyth.network/developers/price-feed-ids).
```

File: docs/tools/oracles/pyth.md [hunk 18]
```diff
@@ -0,0 +1,66 @@
...
+The Pyth Network smart contract is available at the following address: 
+
+- Mainnet: [0xff1a0f4744e8582DF1aE09D5611b887B6a12925C](https://polygonscan.com/address/0xff1a0f4744e8582df1ae09d5611b887b6a12925c).
+- Amoy: [0x2880aB155794e7179c9eE2e38200202908C17B43](https://www.oklink.com/amoy/address/0x2880ab155794e7179c9ee2e38200202908c17b43)
+
+Additionally, click to access the [Pyth price-feed IDs](https://pyth.network/developers/price-feed-ids).
+
+## Other
+
+The Pyth Network provides additional tools to developers like this [TradingView Integration](https://docs.pyth.network/guides/how-to-create-tradingview-charts) or the [Gelato Web3 Functions](https://docs.pyth.network/guides/how-to-schedule-price-updates-with-gelato).
```
Comment (kmurphypolygon) [inline docs/tools/oracles/pyth.md]:
```suggestion
The Pyth network provides additional tools to developers, such as [TradingView Integration](https://docs.pyth.network/guides/how-to-create-tradingview-charts), or the [Gelato web3 functions](https://docs.pyth.network/guides/how-to-schedule-price-updates-with-gelato).  
```

File: docs/tools/oracles/pyth.md [hunk 19]
```diff
@@ -0,0 +1,66 @@
...
+- Mainnet: [0xff1a0f4744e8582DF1aE09D5611b887B6a12925C](https://polygonscan.com/address/0xff1a0f4744e8582df1ae09d5611b887b6a12925c).
+- Amoy: [0x2880aB155794e7179c9eE2e38200202908C17B43](https://www.oklink.com/amoy/address/0x2880ab155794e7179c9ee2e38200202908c17b43)
+
+Additionally, click to access the [Pyth price-feed IDs](https://pyth.network/developers/price-feed-ids).
+
+## Other
+
+The Pyth network provides additional tools to developers, such as [TradingView Integration](https://docs.pyth.network/guides/how-to-create-tradingview-charts), or the [Gelato web3 functions](https://docs.pyth.network/guides/how-to-schedule-price-updates-with-gelato).  
+
+If you have any questions or issues, you can contact us on the following platforms: [Telegram](https://t.me/Pyth_Network), [Discord](https://discord.gg/invite/PythNetwork), [Website](https://pyth.network/contact).
```
Comment (kmurphypolygon) [inline docs/tools/oracles/pyth.md]:
```suggestion
If you have any questions or issues, contact us on the following platforms: 

- [Telegram](https://t.me/Pyth_Network)
- [Discord](https://discord.gg/invite/PythNetwork)
- [Website](https://pyth.network/contact)
```

File: docs/tools/oracles/pyth.md [hunk 20]
```diff
@@ -0,0 +1,70 @@
...
+## Pyth on Polygon
+
+The Pyth Network smart contract is available at the following address: 
+
+- Mainnet: [0xff1a0f4744e8582DF1aE09D5611b887B6a12925C](https://polygonscan.com/address/0xff1a0f4744e8582df1ae09d5611b887b6a12925c).
+- Amoy: [0x2880aB155794e7179c9eE2e38200202908C17B43](https://www.oklink.com/amoy/address/0x2880ab155794e7179c9ee2e38200202908c17b43)
+
+Additionally, click to access the [Pyth price-feed IDs](https://pyth.network/developers/price-feed-ids).
+
+## Other
```
Comment (kmurphypolygon) [inline docs/tools/oracles/pyth.md]:
```suggestion
## Developers and community
```

**Review by kmurphypolygon — APPROVED** [review]

**Pull request merged**